beluga_benchmark/docs/reports/2023-09-02/REPORT.md

```diff
@@ -0,0 +1,122 @@
+- Above $50k$ particles `nav2_amcl` single-threaded process begins to saturate the CPU and its APE metrics begin to deteriorate, while `beluga_amcl` in both configurations remains stable thanks to its lower CPU load. For particle counts below $50k$ the APE metrics of all three alternatives are similar.
+- The multi-threaded configuration of `beluga_amcl` has lower latency than the single-threaded one, at the expense of additional CPU load. The latency of `nav2_amcl` was not measured for the reasons explained above.
+
+#### Beluga vs. Nav2 AMCL using Beam sensor model
+
+In the following plot the results of the benchmark are shown for all three of the tested configurations when using the Beam Sensor model. The vertical scale is logarithmic to better show the differences between the configurations throughout the whole range of particle counts.
+
+![Beluga Seq vs Beluga Par vs. Nav2 AMCL with Beam Sensor Model](beam_beluga_vs_beluga_vs_amcl.png)
+
+Comments on the results:
+
+- `beluga_amcl` in both multi-threaded and single-threaded configurations uses significantly less memory than `nav2_amcl`.
+- Both the single-threaded configuration of `beluga_amcl` and `nav2_amcl` have similar CPU load performance when using the `beam` sensor model. The multi-threaded configuration of `beluga_amcl` uses more CPU than the single-threaded one due to the additional synchronization overhead.
+- Above $50k$ particles both `nav2_amcl` and the single-threaded configuration of `beluga_amcl` begin to saturate the CPU and their APE metrics begin to deteriorate, while `beluga_amcl` in multi-threaded configuration remains stable. For particle counts below $50k$ the APE metrics of all three alternatives are similar.
+- The multi-threaded configuration of `beluga_amcl` has lower latency than the single-threaded one, at the expense of additional CPU load. The latency of `nav2_amcl` was not measured for the reasons explained above.
+
+## Conclusions
+
+- `beluga_amcl`'s memory usage is significantly lower than that of `nav2_amcl` in all configurations.
+- The single-threaded configuration of `beluga_amcl` performs at a lower or equal CPU load than that of `nav2_amcl`. When configured to use the `likelihood` sensor model the performance of the single-threaded `beluga_amcl` configuration is significantly better than that of `nav2_amcl`.
+- The multi-threaded configuration of `beluga_amcl` always uses more CPU than the single-threaded configuration.
+- `beluga_amcl`'s APE performance is similar to that of `nav2_amcl` for lower particle counts. For higher particle counts the performance begins to deteriorate when the processes saturate the available CPU resources.
+- The multi-threaded configuration of `beluga_amcl` has lower latency than the single-threaded one, at the expense of additional CPU load.
```

Comment:
Should we mention that (as expected) the multi-threaded setup remains stable for much higher particle counts?

Reply:
> The fact that the multi-threaded setup (using the beam model) provides an order of magnitude lower latency is great.

Indeed! keep in mind, though, that I'm running this on a 16 cores i9, so our mileage will vary on other platforms.

> Should we mention that (as expected) the multi-threaded setup remains stable for much higher particle counts?

Will do.